lib/js/components/Layouts/ThreeColumnLayout/ThreeColumnLayout.vue

```diff
@@ -0,0 +1,265 @@
+<template>
+	<div class="threeColumnLayout">
+		<div
+			class="threeColumnLayout__overlay"
+			:class="{
+				'-visible': rightColumnVisibleMobile || leftColumnVisibleMobile,
+			}"
+			@click="$emit('overlay-clicked')"
+		/>
+		<div
+			class="threeColumnLayout__leftColumn"
+			:class="{
+				'-desktopVisible': leftColumnVisible || leftColumnVisibleMobile,
+				'-mobileVisible': leftColumnVisibleMobile,
+			}"
+		>
+			<slot name="leftColumn" />
+		</div>
+		<div class="threeColumnLayout__contentColumn">
+			<div
+				class="threeColumnLayout__content"
+				:class="{
+					'-noPadding': contentWithoutPadding,
+				}"
+			>
+				<slot />
+			</div>
+		</div>
+		<div
+			class="threeColumnLayout__rightColumn"
+			:class="{
+				'-medium': rightColumnSize === THREE_COLUMN_LAYOUT_RIGHT_COLUMN_SIZE.MEDIUM,
+				'-large': rightColumnSize === THREE_COLUMN_LAYOUT_RIGHT_COLUMN_SIZE.LARGE,
+				'-desktopVisible': rightColumnVisible || rightColumnVisibleMobile,
+				'-mobileVisible': rightColumnVisibleMobile,
+			}"
+		>
+			<slot name="rightColumn" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/media-queries';
+@import '../../../../styles/settings/z-indexes';
+
+$left-column-width: 23vw;
+$left-column-min-width: 250px;
+$left-column-max-width: 400px;
+
+$content-column-max-width: 900px;
+
+$right-column-min-width: 320px;
+$right-column-medium-max-width: 400px;
+$right-column-medium-l-width: 30vw;
+$right-column-medium-l-min-width: $right-column-min-width;
+$right-column-medium-l-max-width: $right-column-medium-max-width;
+
+$right-column-large-max-width: 500px;
+$right-column-large-l-width: 30vw;
+$right-column-large-l-min-width: $right-column-min-width;
+$right-column-large-l-max-width: $right-column-large-max-width;
+
+.threeColumnLayout {
+	$root: &;
+
+	display: flex;
+	flex-wrap: nowrap;
+	height: 100%;
+	overflow: hidden;
+	position: relative;
+	width: 100%;
+
+	&__overlay {
+		background: $color-default-overlay;
+		display: none;
+		height: 100%;
+		left: 0;
+		position: fixed;
+		top: 0;
+		width: 100%;
+		z-index: $z-index-overlay + 1;
```

Comment:
Why do we need to increase default `z-index` value? Should we add 1 in the variable?